Python/pythonrun.c

```diff
@@ -908,38 +908,32 @@ write_indented_margin(struct exception_print_context *ctx, PyObject *f)
     return _Py_WriteIndentedMargin(EXC_INDENT(ctx), EXC_MARGIN(ctx), f);
 }
 
-static void
-print_exception(struct exception_print_context *ctx, PyObject *value)
+static int
+print_exception_invalid_type(struct exception_print_context *ctx, PyObject *value)
 {
-    int err = 0;
-    PyObject *type, *tb, *tmp;
     PyObject *f = ctx->file;
-
-    _Py_IDENTIFIER(print_file_and_line);
-
-    if (!PyExceptionInstance_Check(value)) {
-        if (err == 0) {
-            err = _Py_WriteIndent(EXC_INDENT(ctx), f);
-        }
-        if (err == 0) {
-            err = PyFile_WriteString("TypeError: print_exception(): Exception expected for value, ", f);
-        }
-        if (err == 0) {
-            err = PyFile_WriteString(Py_TYPE(value)->tp_name, f);
-        }
-        if (err == 0) {
-            err = PyFile_WriteString(" found\n", f);
-        }
-        if (err != 0) {
-            PyErr_Clear();
-        }
-        return;
+    if (_Py_WriteIndent(EXC_INDENT(ctx), f) < 0) {
+        return -1;
+    }
+    if (PyFile_WriteString("TypeError: print_exception(): Exception expected for value, ", f) < 0) {
```

Comment:
PEP 7. Find a nice way to wrap this line.

```suggestion
    const char *const msg = "TypeError: print_exception(): Exception expected "
                            "for value, ";
    if (PyFile_WriteString(msg, f) < 0) {
```